tensorflow_datasets/text/stanford_quad.py

```diff
@@ -0,0 +1,212 @@
+        citation=_CITATION,
+    )
+
+  def _vocab_text_gen(self, filepath):
+    for ex in self._generate_examples(filepath):
+      # "first_answer" is a substring of "context" so not need to add it here
+      yield " ".join([ex["question"], ex["context"]])
+
+  def _split_generators(self, dl_manager):
+    urls_to_download = {
+        "train": os.path.join(self._URL, self._TRAINING_FILE),
+        "dev": os.path.join(self._URL, self._DEV_FILE)
+    }
+    downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+    # Generate shared vocabulary
+    # maybe_build_from_corpus uses SubwordTextEncoder if that's configured
+    self.info.features["context"].maybe_build_from_corpus(
+        self._vocab_text_gen(downloaded_files["train"]))
+    encoder = self.info.features["context"].encoder
+    # Use maybe_set_encoder because the encoder may have been restored from
+    # package data.
+    self.info.features["question"].maybe_set_encoder(encoder)
+    self.info.features["first_answer"].maybe_set_encoder(encoder)
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
+            gen_kwargs={"filepath": downloaded_files["train"]}),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=1,
+            gen_kwargs={"filepath": downloaded_files["dev"]}),
+    ]
+
+  def _generate_examples(self, filepath):
+    """This function returns the examples in the raw (text) form."""
+    logging.info("generating examples from = %s", filepath)
+    with tf.io.gfile.GFile(filepath) as f:
+      squad = json.load(f)
+      for article in squad["data"]:
+        if "title" in article:
+          title = article["title"].strip()
+        else:
+          title = ""
+        for paragraph in article["paragraphs"]:
+          context = paragraph["context"].strip()
+          for qa in paragraph["qas"]:
+            question = qa["question"].strip()
+            id_ = qa["id"]
+
+            answer_starts = [answer["answer_start"] for answer in qa["answers"]]
+            answers = [answer["text"].strip() for answer in qa["answers"]]
+
+            # Features currently used are "context", "question", and "answers".
+            # Others are extracted here for the ease of future expansions.
+            example = {
+                "title": title,
+                "context": context,
+                "question": question,
+                "id": id_,
+                "answer_starts": answer_starts,
+                "answers": answers,
+            }
+            if len(example["answers"]) > 1:
+              yield {
+                  "question": example["question"],
+                  "first_answer": example["answers"][0],
+                  "context": example["context"]
+              }
+            else:
+              example["answers"] = ['No answer is supported by the paragraph']
```

Comment:
We should be as "close" to the raw dataset as possible.

Consider adding the "is_impossible" label (which exists now in 2.0) and in case where there is no solution: can you just return empty string as first_answer ?

Reply:
@cyfra I am not able to return an empty string as it is giving `ValueError: zero-size array to reduction operation minimum which has no identity`

Reply:
Done @cyfra